tests/Unit/Numerical/LinearAlgebra/Test_InvertMatrix.cpp

```diff
@@ -0,0 +1,172 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Numerical/LinearAlgebra/InvertMatrix.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace {
+using generic = tmpl::integral_list<int32_t, 2, 1>;
+using symmetric = tmpl::integral_list<int32_t, 1, 1>;
+
+template <size_t Dim>
+using matrix_ii = Tensor<double, symmetric,
+                         index_list<SpatialIndex<Dim, UpLo::Lo, Frame::Grid>,
+                                    SpatialIndex<Dim, UpLo::Lo, Frame::Grid>>>;
+
+template <size_t Dim>
+using matrix_ij = Tensor<double, generic,
+                         index_list<SpatialIndex<Dim, UpLo::Lo, Frame::Grid>,
+                                    SpatialIndex<Dim, UpLo::Lo, Frame::Grid>>>;
+
+template <size_t Dim>
+using matrix_iA =
+    Tensor<double, generic,
+           index_list<SpatialIndex<Dim, UpLo::Lo, Frame::Grid>,
+                      SpatialIndex<Dim, UpLo::Up, Frame::Inertial>>>;
+
+// Checks that the inverse of a Tensor<double, Symm, index_list<I1, I2>>
+// is a Tensor<double, Symm, index_list<I3, I4>.
```

Comment:
Yes I could do this.. but I'm not sure I understand the motivation. Testing that the inverse Tensor has the expected indices requires exposing the Tensor template parameters, which makes the associated interfaces more complex. Instead of having some complex-looking code in this function and in lines 127-139, the complex-looking code would be in every `verify_inverse_X` function and its call sites.

I will write this as a fixup commit and we can see how we feel about it.